src/riscv/syscall/saveload.rs

```diff
@@ -0,0 +1,147 @@
+// Provides the ability to interact with state storage.
+use std::cell::RefCell;
+use std::cmp;
+use std::rc::Rc;
+
+use ckb_vm::instructions::Register;
+use ckb_vm::Memory;
+use ethereum_types::{Address, H256, U256};
+
+use crate::common::hash;
+use crate::evm::DataProvider;
+use crate::riscv::syscall::common::get_arr;
+use crate::riscv::syscall::convention::{SYSCODE_LOAD, SYSCODE_SAVE};
+
+// Convert k/v of any length to H256 storage.
+//
+// Give an example shows below:
+//
+// k = {0x00, 0x01, 0x02}, len=3
+// v = {0x00, 0x01, 0x02, ... , 0x39}; len=40
+//
+// let k_hash = keccek256(k); // f84a97f1f0a956e738abd85c2ea5026f8874e3ec09c8f12159dfeeaab2b156
+// let v_size = v.len();      // 0x28
+//
+// slot0: 0xf84a97f...56 = 0x........0028 // store the size of v
+// slot1: 0xf84a97f...57 = 0x000102..3031 // store the first 32 bytes of v
+// slot2: 0xf84a97f...58 = 0x32...39...00 // store the rest 8 bytes of v, and fill in with 0x00.
+fn gen_mapped(k: &[u8], v: &[u8]) -> Vec<(Vec<u8>, Vec<u8>)> {
```

Comment:
This is going to be optimized in the future, right?

Reply:
Depends on whether the performance is met. To be honest, I don't really want to do this optimization, It's leading to uncertainty behave.